examples/systemcalls/systemcalls.c

```diff
@@ -93,7 +150,57 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
-    va_end(args);
+    //Open output file
+    int fd = open(outputfile, O_WRONLY | O_CREAT | O_TRUNC, 0644);
+    if(fd == -1)
+    {
+        perror("open");
+	return false;
+    }
+
+    //Create new child process
+    int kidpid = fork();
 
+    switch(kidpid)
+    {
+        case -1:
+            perror("fork");
+            return false;
+	case 0:
+	    if(dup2(fd, 1) < 0)
+	    {
+                perror("dup2");
+		exit(1);
+	    }
+	    close(fd);
+	    execv(command[0], command);
+	    perror("execv");
+	    exit(1);
+	default:
+	    close(fd);
+	    int status;
+	    pid_t wait_pid = waitpid(kidpid, &status, 0);
+	    if(wait_pid == -1)
+	    {
+                perror("wait");
+		return false;
+	    }
+	    else
+	    {
+                if(WIFEXITED(status))
+                {
+                    if(WEXITSTATUS(status) != 0)
+                    {
+                        return false;
```

Comment:
Missing cleanup in these code paths
Also note the discussion in class regarding how to combine logic in these two similar functions.